torchtext/transforms.py

```diff
@@ -189,3 +186,294 @@ def forward(self, input: Any) -> Any:
         :rtype: Union[List[Union[str, int]], List[List[Union[str, int]]]]
         """
         return F.truncate(input, self.max_seq_len)
+
+
+class GPT2BPETokenizer(Module):
+    """
+    Transform for GPT-2 BPE Tokenizer.
+
+    Reimplements openai GPT-2 BPE in TorchScript. Original openai implementation
+    https://github.com/openai/gpt-2/blob/master/src/encoder.py
+
+    :param bpe_encoder: A dict mapping from bpe tokens(e.g sub-words) to
+            bpe token ids. For example: {"ingly": 4420}
+    :type bpe_encoder: Dict[str, int]
+    :param bpe_merge_ranks: A dict mapping a pair of bpe tokens to rank.
+            Torchscript doesn't support tuple as dict key type, concat
+            a pair of string with SEPERATOR (e.g \u0001) as workaround.
+            For example: {"ing\u0001ly": 210}
+    :type bpe_merge_ranks: Dict[str, int]
+    :param byte_encoder: A dict mapping a byte to an unicode character.
+    :type byte_encoder: Dict[int, str]
+    """
+    SEPERATOR: str = "\u0001"
+    LSTRIP_PATTERN: str = "\u0120"
+
+    def __init__(
+        self,
+        encoder_json_path: str,
+        vocab_bpe_path: str,
+    ):
+        super().__init__()
+        self.merge_seperator = self.SEPERATOR
+        self.lstrip_pattern = self.LSTRIP_PATTERN
```

Comment:
nit: can we instead directly call self.SEPERATOR and self.LSTRIP_PATTERN? not sure if it is necessary to assign to another class variable?